source/characterProcessing.py

```diff
@@ -402,82 +405,56 @@ def _saveSymbol(self, symbol):
 		return "\t".join(fields)
 
 
-_noSymbolLocalesCache = set()
-_noCLDRLocalesCache = set()
-
-
-def _getSpeechSymbolsForLocale(locale: str) -> Tuple[SpeechSymbols, SpeechSymbols]:
-	if locale in _noSymbolLocalesCache and (
-		locale in _noCLDRLocalesCache or not config.conf["speech"]["includeCLDR"]
-	):
-		raise LookupError
-	builtinDataImported = False
-	builtin = SpeechSymbols()
-	if config.conf["speech"]["includeCLDR"]:
-		# Try to load CLDR data when processing is on.
-		# Load the data before loading other symbols,
-		# in order to allow translators to override them.
+def _getSpeechSymbolsForLocale(locale: str) -> list[SpeechSymbols]:
+	symbols: list[SpeechSymbols] = []
+	for definition in getAvailableSymbolDictionaryDefinitions():
+		if not definition.enabled:
+			continue
 		try:
-			builtin.load(
-				os.path.join(globalVars.appDir, "locale", locale, "cldr.dic"),
-				allowComplexSymbols=False,
+			symbols.append(definition.getSymbols(locale))
+		except (LookupError, FileNotFoundError):
+			log.debugWarning(
+				f"Error loading {definition.name!r} symbols for locale {locale!r}",
+				exc_info=True,
 			)
-			builtinDataImported = True
-		except IOError:
-			_noCLDRLocalesCache.add(locale)
-			log.debugWarning("No CLDR data for locale %s" % locale)
-	try:
-		builtin.load(os.path.join(globalVars.appDir, "locale", locale, "symbols.dic"))
-		builtinDataImported = True
-	except IOError:
-		_noSymbolLocalesCache.add(locale)
-		log.debugWarning("No symbol data for locale %s" % locale)
-	if not builtinDataImported:
-		raise LookupError("No symbol information for locale %s" % locale)
-	user = SpeechSymbols()
-	pathToSymbolsDic = WritePaths.getSymbolsConfigFile(locale)
-	try:
-		# Don't allow users to specify complex symbols
-		# because an error will cause the whole processor to fail.
-		user.load(pathToSymbolsDic, allowComplexSymbols=False)
-	except IOError:
-		# An empty user SpeechSymbols is okay.
-		pass
-	return builtin, user
+			return symbols
+	if len(symbols) <= 1:
+		raise LookupError(f"No symbol information for locale {locale!r}")
+	return symbols
 
 
-class SpeechSymbolProcessor(object):
+class SpeechSymbolProcessor:
 	"""
 	Handles processing of symbol pronunciation for a locale.
 	Pronunciation information is taken from one or more L{SpeechSymbols} instances.
 	"""
 
 	#: Caches symbol data for locales.
-	localeSymbols: LocaleDataMap[Tuple[SpeechSymbols, SpeechSymbols]] = LocaleDataMap(
```

Comment:
I'm not sure if changing this type is safe outside of an API breaking release

Reply:
I assume a list could be considered tuple compatible. Note that the given list will contain at least two dictionaries, the builtin dictionary and the user dictionary (empty if not available). Loading will result in an error when there's no builtin data.

Reply:
A list is not tuple compatible, they have different class methods. I think this will have to be postponed unless an API break can be avoided